src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -134,7 +137,25 @@ export class FocusMonitor implements OnDestroy {
     this._windowFocusTimeoutId = setTimeout(() => this._windowFocused = false);
   }
 
-  constructor(private _ngZone: NgZone, private _platform: Platform) {}
+   /** Used to reference correct document/window */
+   protected _document?: Document;
+
+  constructor(private _ngZone: NgZone,
+              private _platform: Platform,
+              /** @breaking-change 11.0.0 make document required */
+              @Optional() @Inject(DOCUMENT) document?: any) {
+    this._document = document;
+  }
+
+  /** Access injected document if available or fallback to global document reference */
+  get document(): Document {
+    return this._document || document;
+  }
```

Comment:
We try to only use getters/setters for Angular `@Input` properties (since they're set from the template). For anything else, we prefer a function because getters generate a lot of code for ES5 outputs. 

(here and elsewhere)